### DISCUSSION TITLE
Fully automated HTTPS / TLS setup using Acme

### DISCUSSION THREAD

**Discussion by joepio** [meilisearch/product#609] [Feedback & Feature Proposal]
Meilisearch is incredibly easy to setup. However, bringing it to production requires setting up HTTPS, and that is a huge hassle. Setting up nginx, certbot, moving files... It takes up a [big chunk of your docs](https://docs.meilisearch.com/learn/cookbooks/running_production.html#a-quick-introduction). 

But there's a really nice solution! TLS setup can be almost entirely automated. I've done so myself in [Atomic-Server](https://github.com/atomicdata-dev/atomic-data-rust), which is also a Rust + Actix project, so its implementation could be used as an example.

Read more in [this issue](https://github.com/atomicdata-dev/atomic-data-rust/issues/192) where I compare some crates. I also think [this issue in `instant_acme`](https://github.com/InstantDomain/instant-acme/issues/13) is relevant.

[Implementation example here](https://github.com/atomicdata-dev/atomic-data-rust/blob/3e43a614ad55e68e6eb3e2935e64d939a10232a5/server/src/https.rs), which supports both Http01 and Dns01 handshake. It writes certificates and keys to the filesystem, and deals with renewals.

**Comment by mmachatschek** [comment]
I would recommend replacing Nginx with Caddy Webserver which has automatic HTTPS setup using ACME and additional fallback implementations if the bigger ACME providers are down.

Caddy is easy to setup and easy to understand also for non devops guys

References:

- https://caddyserver.com/
- https://github.com/caddyserver/caddy

**Reply by mmachatschek** [reply]
It also has easy upgrade commands to install newer versions of the webserver:

https://caddyserver.com/docs/command-line#caddy-upgrade

**Comment by macraig** [comment]
Hi @joepio @mmachatschek 👋 Thanks for sharing your ideas with us! We appreciate you taking the time to help us improve our product. Right now, our team is working on a few other projects that are higher up on our priority list. But don't worry, I've added your suggestion to our backlog, and we'll keep it in mind for the future.

Keep the feedback coming, and feel free to reach out if you have any questions or concerns!
